From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kanaka Raju Nayana <kanaka.raju.nayana@intel.com>
Date: Wed, 29 Oct 2025 19:52:07 +0530
Subject: [PATCH] autotools/m4: manage MODULE_IMPORT_NS prototype change

Conditionally add INTEL_PMT namespace import and fix
EXPORT_SYMBOL_NS_GPL double-quoting in generated .mod.c files.

Uses feature detection instead of macro overrides to avoid interfering
with kernel's existing namespace handling.

V1: Rebase to 6.17

V2: fix VSC_TP namespace import for mei-vsc

Reference:
    cdd30ebb1b9f
    module: Convert symbol namespace to string literal

Signed-off-by: Kanaka Raju Nayana <kanaka.raju.nayana@intel.com>
---
 drivers/misc/mei/platform-vsc.c            | 4 ++++
 drivers/platform/x86/intel/pmt/crashlog.c  | 4 ++++
 drivers/platform/x86/intel/pmt/telemetry.c | 5 +++++
 3 files changed, 13 insertions(+)

diff --git a/drivers/misc/mei/platform-vsc.c b/drivers/misc/mei/platform-vsc.c
index 9787b9c..cfc9646 100644
--- a/drivers/misc/mei/platform-vsc.c
+++ b/drivers/misc/mei/platform-vsc.c
@@ -456,4 +456,8 @@ MODULE_AUTHOR("Wentong Wu <wentong.wu@intel.com>");
 MODULE_AUTHOR("Zhifeng Wang <zhifeng.wang@intel.com>");
 MODULE_DESCRIPTION("Intel Visual Sensing Controller Interface");
 MODULE_LICENSE("GPL");
+#ifdef BPM_MODULE_IMPORT_TO_STRING_LITERAL_PRESENT
+MODULE_IMPORT_NS(VSC_TP);
+#else
 MODULE_IMPORT_NS("VSC_TP");
+#endif
diff --git a/drivers/platform/x86/intel/pmt/crashlog.c b/drivers/platform/x86/intel/pmt/crashlog.c
index b0393c9..9a581d3 100644
--- a/drivers/platform/x86/intel/pmt/crashlog.c
+++ b/drivers/platform/x86/intel/pmt/crashlog.c
@@ -611,4 +611,8 @@ module_exit(pmt_crashlog_exit);
 MODULE_AUTHOR("Alexander Duyck <alexander.h.duyck@linux.intel.com>");
 MODULE_DESCRIPTION("Intel PMT Crashlog driver");
 MODULE_LICENSE("GPL v2");
+#ifdef BPM_MODULE_IMPORT_TO_STRING_LITERAL_PRESENT
+MODULE_IMPORT_NS(INTEL_PMT);
+#else
 MODULE_IMPORT_NS("INTEL_PMT");
+#endif
diff --git a/drivers/platform/x86/intel/pmt/telemetry.c b/drivers/platform/x86/intel/pmt/telemetry.c
index a52803b..03d2609 100644
--- a/drivers/platform/x86/intel/pmt/telemetry.c
+++ b/drivers/platform/x86/intel/pmt/telemetry.c
@@ -438,5 +438,10 @@ module_exit(pmt_telem_exit);
 MODULE_AUTHOR("David E. Box <david.e.box@linux.intel.com>");
 MODULE_DESCRIPTION("Intel PMT Telemetry driver");
 MODULE_LICENSE("GPL v2");
+#ifdef BPM_MODULE_IMPORT_TO_STRING_LITERAL_PRESENT
+MODULE_IMPORT_NS(INTEL_PMT);
+MODULE_IMPORT_NS(INTEL_VSEC);
+#else
 MODULE_IMPORT_NS("INTEL_PMT");
 MODULE_IMPORT_NS("INTEL_VSEC");
+#endif
-- 
2.43.0
